City-of-Helsinki/berth-reservations-admin
# VEN-927 | Create additional invoice

## Description :sparkles:
* Create invoice of only one additional product (currently only "storage on ice / season") for specific customer
* Put it behind a feature flag
* Make some invoicing related things reusable as utils

## Issues :bug:

### Closes :no_good_woman:
[VEN-927](https://helsinkisolutionoffice.atlassian.net/browse/VEN-927)
### Related :handshake:

## Testing :alembic:

### Automated tests :gear:️
Several component and util tests in `src/features/createAdditionalInvoice/__tests__`

### Manual testing :construction_worker_man:
Land to:
[http://localhost:3000/customers/UHJvZmlsZU5vZGU6MTBiZGYyYzUtYzk0NS00MGNlLTk0M2QtZjU4YTA3MDU4YzE1](http://localhost:3000/customers/UHJvZmlsZU5vZGU6MTBiZGYyYzUtYzk0NS00MGNlLTk0M2QtZjU4YTA3MDU4YzE1)
You should be able to create the invoice, but the sending doesn't send the invoice because it's pending some BE adjustments, so don't click "Lähetä" for now.

Land to:
[http://localhost:3000/customers/UHJvZmlsZU5vZGU6MDA5NDdlMjctMjZlNS00ZjU5LTg4NWMtYzM1ZWU0ZGJlYjZk](http://localhost:3000/customers/UHJvZmlsZU5vZGU6MDA5NDdlMjctMjZlNS00ZjU5LTg4NWMtYzM1ZWU0ZGJlYjZk)
You should not be able to create the invoice since the customer doesn't have paid berth leases.

## Additional notes :spiral_notepad:
Further adjustments needed in BE related to sending the invoice and in FE to show the invoice correctly in invoice history.
